homeassistant/components/onvif/parsers.py

```diff
@@ -370,6 +370,63 @@ async def async_parse_vehicle_detector(uid: str, msg) -> Event | None:
         return None
 
 
+@PARSERS.register("tns1:RuleEngine/TPSmartEventDetector/TPSmartEvent")
+@PARSERS.register("tns1:RuleEngine/PeopleDetector/People")
+async def async_parse_tplink_detector(uid: str, msg) -> Event | None:
+    """Handle parsing tplink smart event messages.
+
+    Topic: tns1:RuleEngine/TPSmartEventDetector/TPSmartEvent
+    Topic: tns1:RuleEngine/PeopleDetector/People
+    """
+    video_source = ""
+    video_analytics = ""
+    rule = ""
+    topic = ""
+    vehicle = False
+    person = False
+    enabled = False
+    try:
+        topic, payload = extract_message(msg)
+        for source in payload.Source.SimpleItem:
+            if source.Name == "VideoSourceConfigurationToken":
+                video_source = _normalize_video_source(source.Value)
+            if source.Name == "VideoAnalyticsConfigurationToken":
+                video_analytics = source.Value
+            if source.Name == "Rule":
+                rule = source.Value
+
+        for item in payload.Data.SimpleItem:
+            if item.Name == "IsVehicle":
+                vehicle = True
+                enabled = item.Value == "true"
+            if item.Name == "IsPeople":
+                person = True
+                enabled = item.Value == "true"
+    except (AttributeError, KeyError):
+        return None
+
+    if vehicle:
+        return Event(
+            f"{uid}_{topic}_{video_source}_{video_analytics}_{rule}",
+            "Vehicle Detection",
+            "binary_sensor",
+            "motion",
+            None,
+            enabled,
+        )
+    if person:
+        return Event(
+            f"{uid}_{topic}_{video_source}_{video_analytics}_{rule}",
+            "Person Detection",
+            "binary_sensor",
+            "motion",
+            None,
+            enabled,
+        )
+
+    return None
```

Comment:
It looks like we are missing coverage for this line

Reply:
Added a couple tests to improve coverage.